### PR TITLE
feat(plugins): PluginConfiguration annotation

### PR DESCRIPTION
**Pull request by jonsie**
This removes `@ExtensionConfiguration`.  If this general idea seems to make sense I'll clean this up and fix up docs and tests.

### CODE REVIEW

**Comment by robzienert** [comment]
Do like. Let's move forward with this.

**Comment by jonsie** [comment]
> Do like. Let's move forward with this.

I'll do this today.

File: kork-plugins/src/test/kotlin/com/netflix/spinnaker/kork/plugins/update/release/source/SpringPluginInfoReleaseSourceTest.kt [hunk 1]
```diff
@@ -62,8 +62,8 @@ class SpringPluginInfoReleaseSourceTest : JUnit5Minutests {
 
       expectThat(releases).isA<Set<PluginInfoRelease>>()
         .get { releases.size }.isEqualTo(2)
-        .get { releases.find { it?.pluginId == plugin1.id } }.isEqualTo(PluginInfoRelease(plugin1.id, plugin1ExpectedRelease))
-        .get { releases.find { it?.pluginId == plugin2.id } }.isEqualTo(PluginInfoRelease(plugin2.id, plugin2ExpectedRelease))
+        .get { releases.find { it.pluginId == plugin1.id } }.isEqualTo(PluginInfoRelease(plugin1.id, plugin1ExpectedRelease))
+        .get { releases.find { it.pluginId == plugin2.id } }.isEqualTo(PluginInfoRelease(plugin2.id, plugin2ExpectedRelease))
```
Comment (jonsie) [inline kork-plugins/src/test/kotlin/com/netflix/spinnaker/kork/plugins/update/release/source/SpringPluginInfoReleaseSourceTest.kt:66]: Just cleaning up some warnings.

File: kork-plugins/src/test/kotlin/com/netflix/spinnaker/kork/plugins/config/SpringEnvironmentConfigResolverTest.kt [hunk 2]
```diff
@@ -74,7 +74,7 @@ class SpringEnvironmentConfigResolverTest : JUnit5Minutests {
           ExtensionConfigCoordinates("netflix.sweet-plugin", "config.nonexistent"),
           MissingNoArgConstructor::class.java
         )
-      }.succeeded()
+      }.isSuccess()
```
Comment (jonsie) [inline kork-plugins/src/test/kotlin/com/netflix/spinnaker/kork/plugins/config/SpringEnvironmentConfigResolverTest.kt:77]: `succeeded` is now deprecated, replaced with `isSuccess`.

**Comment by jonsie** [comment]
@robzienert This is ready.

I decided to keep the existing `ExtensionConfiguration` functionality for now, but mark it as deprecated.  We have a fair number of plugins that use `ExtensionConfiguration` and I want to give us a time window to migrate.

File: kork-plugins-api/src/main/java/com/netflix/spinnaker/kork/plugins/api/PluginConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+import com.netflix.spinnaker.kork.annotations.Alpha;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Denotes that a class provides plugin configuration. Classes annotated with @PluginConfiguration
```
Comment (robzienert) [inline kork-plugins-api/src/main/java/com/netflix/spinnaker/kork/plugins/api/PluginConfiguration.java:28]:
```suggestion
 * Denotes that a class provides plugin configuration. Classes annotated with [PluginConfiguration]
```

**Pull request merged**